test/blackbox-tests/mel-unwrap-in-jsproperties.t

```diff
@@ -0,0 +1,73 @@
+Tests for deriving `jsProperties, getSet`
+
+  $ . ./setup.sh
+
+`[@@deriving jsProperties, getSet]` respects `[@mel.unwrap]`
+
+  $ cat > x.ml <<EOF
+  > type t = {
+  >   mutable action : [ \`String of string | \`Int of int ] option;
+  >       [@mel.unwrap] [@mel.optional]
+  >   other : [ \`String of int | \`Int of string ] option; [@mel.optional]
+  > }
+  > [@@deriving jsProperties, getSet]
```

Comment:
Discussed with @anmonteiro offline. The reason why `getSet` is needed is that regular / manual mutations of the field will be unsound / unsafe.

I suggested that we can check if `mel.unwrap` is being used in a `mutable` field and error out at parse time to prevent this. It would limit the functionality but probably avoid some confusion (or worse, runtime exns)

Reply:
I'm actually thinking: what does it mean to use a getter on a field that has `@mel.unwrap` etc?

I think it'll just be unsound based on the object that has been created with a maker 🤔

Reply:
https://github.com/melange-re/melange/pull/1162/commits/ad63bc5349e84503dbdb298c81f017b4d5a43cd1 shows that this feature is unsound even for `<field>Get`, which makes me think that it's not something we can ship